Source/NonlinearSolvers/WarpX_PETSc.H

```diff
@@ -0,0 +1,140 @@
+/* Copyright 2025 Debojyoti Ghosh
+ *
+ * This file is part of WarpX.
+ *
+ * License: BSD-3-Clause-LBNL
+ */
+
+#ifndef WARPX_PETSC_H_
+#define WARPX_PETSC_H_
+
+#include "LinearFunction.H"
+
+#include <AMReX_BLassert.H>
+#include <AMReX_REAL.H>
+#include <AMReX_Config.H>
+
+#ifdef AMREX_USE_PETSC
+
+// forward declarations
+class WarpXSolverVec;
+class ImplicitSolver;
+// end of forward declarations
+
+namespace warpx_petsc {
+
+// forward declarations
+class KSPObj;
+class MatObj;
+class VecObj;
+// end of forward declarations
+
+using LinOpType = LinearFunction<WarpXSolverVec,ImplicitSolver>;
+using VecType = WarpXSolverVec;
+
+class KSP_impl
+{
+    public:
+
+        //! Constructor
+        KSP_impl (LinOpType&);
+
+        //! Destructor
+        ~KSP_impl ();
+
```

Comment:
Should you delete the copy and move constructors and assignment operators, as elsewhere in this PR?